Add more tests, fix bug in add_to_ld_library_path and typos

Flamefire opened the pull request:
I've noticed a few typos and places where the docstrings got outdated which I fixed.

There was also a bug in add_to_ld_library_path where the parent path of an existing path would not be added. Not sure if this can come up in practice but better be safe. I added a test which reproduces this and now passes.

I also made the tests run with every instrumenter (2 by now, but will be more) to make sure all work for all test cases.

The dict change is removing redundant `is in` checks in favor of `get` which accepts a default value

AndreasGocht commented:
Technically this would have been 3 PRs again. One for the typos, one for the bugfix and one for the modifications in the test. However, leave it as it is.

AndreasGocht commented:
There is one question for the mpi part left:

> Just out of interest: Why testing against instrumentation2 as well?

Flamefire commented on test/cases/mpi.py:
```diff
@@ -3,6 +3,7 @@
 from mpi4py import MPI
 import numpy as np
 import mpi4py
+import instrumentation2
```
Looks like I forgot or deleted the test assertions. The intention was to check the generated trace for the calls to bar and baz to make sure the instrumenters get those.

Flamefire replied:
I added this in https://github.com/score-p/scorep_binding_python/pull/93/commits/fed67f1eab19aef3cc3dfe617c71695d4e2b7d73 and found why it wasn't done yet: The imports create about 120k events and as the whole trace file is read into Python via otf2-print this takes forever (~5min for all tests). I changed it slightly so the instrumenter is only enabled after the imports which cuts down on the time significantly.

Flamefire commented:
Rebased to resolve the conflict. Changed/new commits:
- https://github.com/score-p/scorep_binding_python/pull/93/commits/bcd58e3ea35acc8521591970264f4b82a09641d7 to incorporate the CInstrumenter
- https://github.com/score-p/scorep_binding_python/pull/93/commits/7ce775689db6e25e98f3650628b73831fdb3d81c to skip those explicitely and not silently
- https://github.com/score-p/scorep_binding_python/pull/93/commits/fed67f1eab19aef3cc3dfe617c71695d4e2b7d73 for the MPI case
